pvlib/iotools/crn.py

```diff
@@ -0,0 +1,101 @@
+"""Functions to read data from the US Climate Reference Network (CRN).
+"""
+
+import pandas as pd
+import numpy as np
+from numpy import dtype
+
+
+HEADERS = 'WBANNO UTC_DATE UTC_TIME LST_DATE LST_TIME CRX_VN LONGITUDE LATITUDE AIR_TEMPERATURE PRECIPITATION SOLAR_RADIATION SR_FLAG SURFACE_TEMPERATURE ST_TYPE ST_FLAG RELATIVE_HUMIDITY RH_FLAG SOIL_MOISTURE_5 SOIL_TEMPERATURE_5 WETNESS WET_FLAG WIND_1_5 WIND_FLAG'  # noqa: E501
+
+VARIABLE_MAP = {
+    'LONGITUDE': 'longitude',
+    'LATITUDE': 'latitude',
+    'AIR_TEMPERATURE': 'temp_air',
+    'SOLAR_RADIATION': 'ghi',
+    'SR_FLAG': 'ghi_flag',
+    'RELATIVE_HUMIDITY': 'relative_humidity',
+    'RH_FLAG': 'relative_humidity_flag',
+    'WIND_1_5': 'wind_speed',
+    'WIND_FLAG': 'wind_speed_flag'
+}
+
+# as specified in CRN README.txt file. excludes 1 space between columns
+WIDTHS = [5, 8, 4, 8, 4, 6, 7, 7, 7, 7, 6, 1, 7, 1, 1, 5, 1, 7, 7, 5, 1, 6, 1]
+# add 1 to make fields contiguous (required by pandas.read_fwf)
+WIDTHS = [w + 1 for w in WIDTHS]
+# no space after last column
+WIDTHS[-1] -= 1
+
+# specify dtypes for potentially problematic values
+DTYPES = [
+    dtype('int64'), dtype('int64'), dtype('int64'), dtype('int64'),
+    dtype('int64'), dtype('int64'), dtype('float64'), dtype('float64'),
+    dtype('float64'), dtype('float64'), dtype('float64'),
+    dtype('int64'), dtype('float64'), dtype('O'), dtype('int64'),
+    dtype('float64'), dtype('int64'), dtype('float64'),
+    dtype('float64'), dtype('int64'), dtype('int64'), dtype('float64'),
+    dtype('int64')
+]
```

Comment:
nit: why not list comprehension?

```python
DTYPES = [np.dtype(t) for t in (
    # not necessary to use addition and multiplication here if undesired for readability
    ['int64']*6 + ['float64']*5
    + ['int64', 'float64', 'O', 'int64', 'float64', 'int64', 'float64', 'float64', 'int64', 'int64',
       'float64', 'int64'])]
```

Reply:
turns out that you can supply a list of strings and the `dtype` function is not necessary.